contracts/CO30_Tobias_WhiskeyTangoFoxtrot.vt7/init.sqf

```diff
@@ -0,0 +1,325 @@
+] call ace_interact_menu_fnc_createAction;
+[van_1, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+private _action = [
+  "open_rear_doors", // unique id
+  "Open rear doors", // text
+  "", // icon
+  { 
+      van_1 animateDoor ["Door_4_source", 1];
+      missionNamespace setVariable ["rearDoorClosed", false, false];
+      missionNamespace setVariable ["hack", false, false]; 
+  }, // statement
+  { rearDoorClosed } // condition
+] call ace_interact_menu_fnc_createAction;
+[van_1, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+private _action = [
+  "close_rear_doors", // unique id
+  "Close rear doors", // text
+  "", // icon
+  {
+      van_1 animateDoor ["Door_4_source", 0];
+      missionNamespace setVariable ["rearDoorClosed", true, true]; 
+  }, // statement
+  { !rearDoorClosed } // condition
+] call ace_interact_menu_fnc_createAction;
+[van_1, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+//////// Terminal //////////
+
+private _action = [
+    "signal_check", // unique id
+    "Check signal", // text
+    "", // icon
+    {
+        private _termDist1 = terminal_1 distance target_1;
+        private _termDist2 = terminal_1 distance target_2;
+        private _termDist3 = terminal_1 distance target_3;
+
+        private _bestSignal = _termDist1 min _termDist2 min _termDist3;
+        private _signalText = switch (true) do {
+            case (_bestSignal < 250): {
+                "Signal strong"
+            };
+            case (_bestSignal < 500): {
+                "Signal weak"
+            };
+            default {
+                "No signal"
+            };
+        };
+
+        [ ["", _signalText, 0] ] spawn BIS_fnc_EXP_camp_playSubtitles;
+
+        private _runHack1 = _termDist1 < 250;
+        if (_runHack1 != (missionNamespace getVariable ["runHack1", false])) then {
+            missionNamespace setVariable ["runHack1", _runHack1, true];
+        };
+        private _runHack2 = _termDist2 < 250;
+        if (_runHack2 != (missionNamespace getVariable ["runHack2", false])) then {
+            missionNamespace setVariable ["runHack2", _runHack2, true];
+        };
+        private _runHack3 = _termDist3 < 250;
+        if (_runHack3 != (missionNamespace getVariable ["runHack3", false])) then {
+            missionNamespace setVariable ["runHack3", _runHack3, true];
+        };
+    }, // statement
+    { !doorClosed } // condition
+] call ace_interact_menu_fnc_createAction;
+[terminal_1, 0, ["ACE_MainActions"], _action] call ace_interact_menu_fnc_addActionToObject;
+
+//////// Hacking Target 1 //////////
+
+private _action = [
+    "start_hack_1", // unique id
+    "Breach security protocols", // text
+    "", // icon
+    {
+      [
+        10, // Duration
+        [], // Condition
+        { 
+          [
+	          ["", "Breached security protocols", 0]
+              ] spawn BIS_fnc_EXP_camp_playSubtitles;
```

Comment:
I am generally against subtitles / on screen text

The players should be able to tell from the progress bar that they are done the action

We also have the "Upload" script in the synixe mission functions if that can be used at all here, it's more immersive than a progress bar or on screen text